ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/LabelClickSpikes.java

```diff
@@ -46,11 +46,11 @@
   /**
    * Composite transform that wraps {@code DetectClickSpikes} with keying by {@code context_id}.
    */
-  public static PTransform<PCollection<PubsubMessage>, PCollection<PubsubMessage>> perContextId(
+  public static PTransform<PCollection<SponsoredInteraction>, PCollection<SponsoredInteraction>> perContextId(
       Integer maxClicks, Duration windowDuration) {
     return PTransform.compose("DetectClickSpikesPerContextId", input -> input //
-        .apply(WithKeys.of((message) -> message.getAttribute(Attribute.CONTEXT_ID))) //
-        .setCoder(KvCoder.of(StringUtf8Coder.of(), PubsubMessageWithAttributesCoder.of()))
+        .apply(WithKeys.of((interaction) -> interaction.contextID())) //
+        .setCoder(KvCoder.of(StringUtf8Coder.of(), SponsoredInteractionCoder.of()))
```

Comment:
Assuming we get AutoValueSchema working, then I think we'd look up the coder here using the registry per https://beam.apache.org/documentation/programming-guide/#default-coder-lookup